### PR TITLE
runtime: fix channels on AVR

### PR DESCRIPTION
**Pull request by aykevl**
This commit makes channel code compilable on AVR, for when goroutines are supported on that platform. I have in fact managed to get initial support for channels working with this patch on a local branch that adds support for the stack-based scheduler.

### CODE REVIEW

**Comment by aykevl** [comment]
I think this is slightly wrong (but close enough to correct that it should be fine). I will look into fixing this properly, in a similar way as slice bounds were fixed.

**Comment by aykevl** [comment]
Fixed it more universally. Unfortunately, it appears that CircleCI builds are now broken.

**Comment by aykevl** [comment]
Rebased to fix CI (after #950 got merged).

File: src/runtime/panic.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ func slicePanic() {
 	runtimePanic("slice out of range")
 }
 
+// Panic when trying to create a new channel that is too big.
+func chanMakePanic() {
+	runtimePanic("new channel too big")
```
Comment (jaddr2line) [inline src/runtime/panic.go]:
```suggestion
	runtimePanic("new channel is too big")
```

**Comment by aykevl** [comment]
Updated.

**Comment by deadprogram** [comment]
Merging.

**Pull request merged**